frontend/src/lib/components/TaxonomicFilter/infiniteListLogic.ts

```diff
@@ -162,6 +162,9 @@ export const infiniteListLogic = kea<infiniteListLogicType>({
                     }
                 },
                 updateRemoteItem: ({ item }) => {
+                    // On updating item, invalidate cache
+                    apiCache = {}
+                    apiCacheTimers = {}
```

Comment:
I don't think anything will break, but this introduces subtle bugs. The right thing to do would be to also invalidate all existing timers.

Regarding "separation of concerns", the logic now knows way too much about the inner workings of the caching layer. I suggest a `clearCache()` function that just does all of this.